src/jennifer/migration/runner.cr

```diff
@@ -3,19 +3,33 @@ module Jennifer
     module Runner
       MIGRATION_DATE_FORMAT = "%Y%m%d%H%M%S%L"
 
-      def self.migrate(count)
+      def self.adapter(config_key)
+        Jennifer::Adapter::AdapterRegistry.adapter(config_key)
+      end
+
+      def self.migrate(config_key = "default", count = 1)
         performed = false
-        Adapter.adapter.ready_to_migrate!
-        return if ::Jennifer::Migration::Base.migrations.empty?
+        puts "Attempting to migrate using config '#{config_key}'"
+        adapter(config_key).ready_to_migrate!
+
+        if ::Jennifer::Migration::Base.migrations.empty?
+          puts "No migrations found."
+          return
+        end
+
         interpolation = {} of String => typeof(Base.migrations[0])
         Base.migrations.each { |m| interpolation[m.version] = m }
 
         pending = interpolation.keys - Version.all.pluck(:version).map(&.as(String))
-        return if pending.empty?
-        brocken = Version.where { _version.in(pending) }.pluck(:version).map(&.as(String))
-        unless brocken.empty?
+        if pending.empty?
+          puts "There are no pending migrations"
+          return
+        end
+
+        broken = Version.where { _version.in(pending) }.pluck(:version).map(&.as(String))
```

Comment:
will need to be able to specify an adapter dynamically here to use an adapter other than the default one. It makes sense here but not sure its something we should necessarily promote for models in general. Might be worth pushing this code back up into Version and have it take an adapter as a parameter. Also adapter selection via using_config will need to be introduced into migration base.

Reply:
anyway user is able to reopen class if needed. also we can make just macro which will expand to overriding `Version` class. But I guess we can make defining default connection "must have"